File: src/test/java/no/nav/veilarbaktivitet/aktivitetskort/EksternaktivitetDAOTest.java
package no.nav.veilarbaktivitet.aktivitetskort;


import lombok.SneakyThrows;
import no.nav.veilarbaktivitet.aktivitet.AktivitetDAO;
import no.nav.veilarbaktivitet.aktivitet.domain.AktivitetData;
import no.nav.veilarbaktivitet.config.database.Database;
import no.nav.veilarbaktivitet.mock.LocalH2Database;
import no.nav.veilarbaktivitet.testutils.AktivitetDataTestBuilder;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.jdbc.core.JdbcTemplate;

class EksternaktivitetDAOTest {

    @Test
    @SneakyThrows
    void test_json_serialisering() {
        JdbcTemplate jdbcTemplate = LocalH2Database.getPresistentDb();
        Database database = new Database(jdbcTemplate);
        AktivitetDAO aktivitetDAO = new AktivitetDAO(database.getNamedJdbcTemplate());

        AktivitetData aktivitetData = AktivitetDataTestBuilder.nyEksternAktivitet();
        AktivitetData opprettetAktivitetData = aktivitetDAO.opprettNyAktivitet(aktivitetData);

        AktivitetData aktivitet = aktivitetDAO.hentAktivitet(opprettetAktivitetData.getId());
        Assertions.assertThat(aktivitet.getEksternAktivitetData()).isEqualTo(opprettetAktivitetData.getEksternAktivitetData());
        Assertions.assertThat(aktivitet.withEksternAktivitetData(null)).isEqualTo(opprettetAktivitetData.withEksternAktivitetData(null));
    }

}

File: src/test/kotlin/no/nav/veilarbaktivitet/aktivitetskort/EksternaktivitetDAOTest.kt
package no.nav.veilarbaktivitet.aktivitetskort

import lombok.SneakyThrows
import no.nav.veilarbaktivitet.LocalDatabaseSingleton
import no.nav.veilarbaktivitet.aktivitet.AktivitetDAO
import no.nav.veilarbaktivitet.testutils.AktivitetDataTestBuilder
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate


internal class EksternaktivitetDAOTest {

    private val dataSource = LocalDatabaseSingleton.postgres

    var jdbc = NamedParameterJdbcTemplate(dataSource)

    @Test
    @SneakyThrows
    fun test_json_serialisering_av_eksternaktivitet() {
        val aktivitetDAO = AktivitetDAO(jdbc)

        val aktivitetData = AktivitetDataTestBuilder.nyEksternAktivitet()
        val opprettetAktivitetData = aktivitetDAO.opprettNyAktivitet(aktivitetData)
        val aktivitet = aktivitetDAO.hentAktivitet(opprettetAktivitetData.id)
        val utEkstern = aktivitet.eksternAktivitetData
        val innEkstern = opprettetAktivitetData.eksternAktivitetData

        assertThat(utEkstern.endretTidspunktKilde).isEqualToIgnoringNanos(innEkstern.endretTidspunktKilde)
        assertThat(utEkstern).isEqualTo(innEkstern.copy(endretTidspunktKilde = utEkstern.endretTidspunktKilde))
        assertThat(aktivitet.withEksternAktivitetData(null)).isEqualTo(opprettetAktivitetData.withEksternAktivitetData(null))
    }
}
